File: URI Online Judge/1164 - Perfect Number.java
import java.util.Scanner;

class Main{
    public static void main(String args[]){
        Scanner inp = new Scanner(System.in);
        int T = inp.nextInt();
        while(T>0){
            int sum=0, n=inp.nextInt();
            for(int a=1;a<n;a++){
                if(n%a==0){
                    sum+=a;
                    if(sum>n){
                        break;
                    }
                }
            }
            if(sum==n)
                System.out.print(n+" eh perfeito\n");
            else
                System.out.print(n+" nao eh perfeito\n");
            T--;
        }
    }
}

File: URI Online Judge/1164 - Perfect Number.cpp
#include <bits/stdc++.h>
using namespace std;

int main()
{
    int T;
    long long n;
    scanf("%d", &T);
    while (T--) {
        scanf("%lld", &n);
        long long sum = 0;
        for (long long a = 1; a < n; a++)
            if (n % a == 0) {
                sum += a;
                if (sum > n) {
                    printf("%d nao eh perfeito\n", n);
                    break;
                }
            }
        if (sum == n)
            printf("%d eh perfeito\n", n);
        else if (sum < n)
            printf("%d nao eh perfeito\n", n);
    }
}